tree.go

```diff
@@ -567,53 +566,81 @@ func (n *InternalNode) Commitment() *Point {
 	return n.commitment
 }
 
+func (n *InternalNode) fillLevels(levels [][]*InternalNode) {
+	levels[int(n.depth)] = append(levels[int(n.depth)], n)
+	for _, child := range n.children {
+		if childInternalNode, ok := child.(*InternalNode); ok && len(childInternalNode.cow) > 0 {
+			childInternalNode.fillLevels(levels)
+		}
+	}
+}
+
 func (n *InternalNode) Commit() *Point {
```

Comment:
I think the Github diffs might be confusing, may be better to see the code change locally or with split-views.